src/meta/client/MetaClient.cpp

```diff
@@ -39,8 +39,24 @@ MetaClient::~MetaClient() {
 }
 
 
-void MetaClient::init() {
+void MetaClient::waitForMetadReady() {
     loadDataThreadFunc();
+
+    if (!ready_ && sendHeartBeat_) {
+        while (true) {
+            if (listener_ == nullptr) {
+                VLOG(1) << "Can't send heartbeat due to listener_ is nullptr!";
```

Comment:
Function “heartbeat” call “auto localHost = listener_->getLocalHost();”, so need continue here

Reply:
fixed! Thanks! But I think exit(EXIT_FAILURE) will be more reasonable!

Reply:
> fixed! Thanks! But I think exit(EXIT_FAILURE) will be more reasonable!

so that mean metaclient is strong depend listener?

Reply:
Now, the logic has changed. Metaclient don't strong depend on listener!